http3/server.go

```diff
@@ -492,14 +527,25 @@ func (s *Server) handleConn(conn quic.Connection) error {
 	// Process all requests immediately.
 	// It's the client's responsibility to decide which requests are eligible for 0-RTT.
 	for {
-		str, datagrams, err := hconn.acceptStream(context.Background())
+		str, datagrams, err := hconn.acceptStream(s.closeCtx)
 		if err != nil {
+			// server is closed, send GOAWAY frame and return, since those requests will be rejected anyway
+			if errors.Is(err, context.Canceled) {
```

Comment:
Probably more error-prone, in case we decide to use a different (or a derived) context here at some point in the future:
```suggestion
			if s.closeCtx.Err() != nil {
```